src/components/calcite-pick-list/shared-list-logic.ts

```diff
@@ -30,6 +31,7 @@ export const sharedListMethods = {
   initialize(this: CalcitePickList | CalciteValueList): void {
     this.setUpItems();
     this.setUpFilter();
+    this.emitCalciteListChangeEventDebounced = debounce(this.emitCalciteListChangeEvent, 10);
```

Comment:
Debouncing seems like a workaround to an underlying problem. What's causing the multiple event firing?

Reply:
The "List" event listens for anytime a "ListChild" selected property changes.

In order to act like a radio button (new selection removes old ones) or when multiple items are selected at the same time, there are multiple selection changes happening which results in multiple "List" event firings.

@pr3tori4n was saying that there should only be one event firing even if multiple items were changed.

Reply:
Chatted w/ @driskull and this is here to batch all change events within a single tick caused by the internal item updating. Based on that, waiting to notify until the next tick would do the job.